### PR TITLE
Servantify the DELETE /self endpoint

### PR DESCRIPTION
**Pull request by mdimjasevic**
This patch rewrites the `DELETE /self` endpoint to Servant.

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information:
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: libs/wire-api/src/Wire/API/Routes/Public/Brig.hs [hunk 1]
```diff
@@ -16,16 +15,19 @@
 --
 -- You should have received a copy of the GNU Affero General Public License along
 -- with this program. If not, see <https://www.gnu.org/licenses/>.
+{-# OPTIONS_GHC -fno-warn-orphans #-}
```
Comment (mdimjasevic) [inline libs/wire-api/src/Wire/API/Routes/Public/Brig.hs:18]: I don't like this so if you have any suggestions where to put that instance for `AsUnion DeleteSelfResponses (Maybe Timeout)`, I am all ears.

File: libs/types-common/src/Data/Code.hs [hunk 2]
```diff
@@ -95,20 +95,27 @@ instance ToHttpApiData Value where
...
 -- | A 'Timeout' is rendered as an integer representing the number of seconds remaining.
 instance ToByteString Timeout where
   builder (Timeout t) = builder (round t :: Int32)
 
 -- | A 'Timeout' is rendered in JSON as an integer representing the
 -- number of seconds remaining.
-instance ToJSON Timeout where
-  toJSON (Timeout t) = toJSON (round t :: Int32)
+instance A.ToJSON Timeout where
+  toJSON (Timeout t) = A.toJSON (round t :: Int32)
```
Comment (pcapriotti) [inline libs/types-common/src/Data/Code.hs:113]: Remove this instance and derive it `via Schema Timeout`.

File: libs/types-common/src/Data/Code.hs [hunk 3]
```diff
@@ -95,20 +95,27 @@ instance ToHttpApiData Value where
...
 
 -- | A 'Timeout' is parsed from JSON as an integer representing the
 -- number of seconds remaining.
-instance FromJSON Timeout where
-  parseJSON = withScientific "Timeout" $ \n ->
+instance A.FromJSON Timeout where
+  parseJSON = A.withScientific "Timeout" $ \n ->
```
Comment (pcapriotti) [inline libs/types-common/src/Data/Code.hs:118]: Same as above.

File: services/brig/src/Brig/API/Error.hs [hunk 4]
```diff
@@ -37,16 +37,16 @@ import Network.HTTP.Types.Header
 import Network.HTTP.Types.Status
 import qualified Network.Wai.Utilities.Error as Wai
 import Servant.API.Status
-import Wire.API.ErrorDescription
+import qualified Wire.API.ErrorDescription as ErrDesc
```
Comment (pcapriotti) [inline services/brig/src/Brig/API/Error.hs:40]: Why?

**Review by pcapriotti — APPROVED** [review]
Looks good! Minor comments follow.

File: services/brig/src/Brig/API/Error.hs (hunk 4 above)
Comment (mdimjasevic) [reply services/brig/src/Brig/API/Error.hs:40]: I introduced a few `ErrorDescription`s with the same names as their counterparts in this module. Should I handle this in some other way?

File: libs/types-common/src/Data/Code.hs (hunk 2 above)
Comment (mdimjasevic) [reply libs/types-common/src/Data/Code.hs]: Done in 5413dd1.

File: libs/types-common/src/Data/Code.hs (hunk 3 above)
Comment (mdimjasevic) [reply libs/types-common/src/Data/Code.hs]: Done in 5413dd1.

**Pull request merged**